### DISCUSSION TITLE
catching auto test crashes with sentry

### DISCUSSION THREAD

**Discussion by bmantas** [getsentry/sentry-unreal#340] [Q&A] [answered]
Hello,

If I understand correctly, Sentry is a game instance subsystem. So, if we run auto tests on CI that don’t create a UGameInstance, there is no way to use Sentry for crash reporting. If that’s the case, are there any plans to have Sentry as, let’s say, a UUnrealEditorSubsystem so we could report editor crashes (including auto tests) with Sentry? I hope my question makes sense. I am rather new to UE.

Thank you!

**Comment by tustanivsky** [comment, marked as answer]
Hey @bmantas, currently Sentry plugin focuses only on "in-game" crash capturing, so tracking those in the scenario you've described above is not possible. We'll definitely take a closer look at this in order to understand whether it's possible to separate the core plugin's logic from `UGameInstanceSubsystem` though I can't provide any ETA when exactly that's going to happen.

**Comment by tustanivsky** [comment]
Probably it won't be very helpful however you might want to check the [SentrySubsystem.spec.cpp](https://github.com/getsentry/sentry-unreal/blob/main/plugin-dev/Source/Sentry/Private/Tests/SentrySubsystem.spec.cpp) implementation we're using for internal testing.
